doc/before-after-reboot-checks.md

```diff
@@ -0,0 +1,46 @@
+# Before and After Reboot Checks
+
+CLUO can require custom node annotations before a node is allowed to reboot or
+before a node is allowed to become schedulable after a reboot. 
+
+## Configuring the `update-operator`
+
+When depolying the `update-operator`, you can provide a comma-separated list of
+node annotations to look for before a reboot should occur, and another one for
+after a reboot has occured. The list is provided as a command line argument to
+the binary -
+
+```bash
+update-operator --before-reboot-annotations anno1,anno2 --after-reboot-annotations anno3,anno4
+```
+
+## Before and After Reboot Labels
+
+The `update-operator` labels nodes that are about to reboot with
+`container-linux-update.v1.coreos.com/before-reboot=true` and labels nodes which
+have just completed rebooting (but are not yet marked as scheduable) with
+`container-linux-update.v1.coreos.com/after-reboot=true`. If you've required
+before or after reboot annotations, the `update-operator` will wait until all
+the respective annotations are applied before proceeding.
+
+## Making a Custom Check
+
+A check that needs to run before reboot should look for the
```

Comment:
Hm, starting with a verb here might hint the user needs to go write some code or pick something. what about:

Write your logic to perform custom before-reboot behavior and after-reboot behavior. When successful, ensure your code sets the annotations you've passed to `update-operator`. When your logic finds an issue, leaving the annotations unset will ensure cluster upgrades halt at the problematic node for a user to intervene.

It is recommended that custom checks be implemented by a container image and deployed using a Daemonset with a node selector on the before-reboot or after-reboot labels. Be sure your image can handle being rescheduled to a node on which it has previously run as the update-operator does not remove the before-reboot and after-reboot labels instantaneously.